vyper/semantics/analysis/imports.py

```diff
@@ -224,6 +214,16 @@ def _load_import(
         search_paths = self.input_bundle.search_paths.copy()  # noqa: F841
         raise ModuleNotFound(module_str, hint=hint) from err
 
+    def _load_file(self, path: PathLike, level: int):
+        if level == 0:
+            self.input_bundle.search_paths = self.absolute_search_paths
+        else:
+            ast = self.graph.current_module
+            current_search_path = Path(ast.resolved_path).parent
+            self.input_bundle.search_paths = [current_search_path]
```

Comment:
i think this corrupts the input bundle's search path, which might be important if anybody tries to access the search path after compilation (for instance as is done in `output_bundle.py`). i think we need to restore the original `.search_paths` after `load_file` is called.